Add arrow

josedlpozo opened the pull request:
Hello, again!

In this PR, I have added the back arrow in MovieDetailsActivity.

![arrow](https://cloud.githubusercontent.com/assets/6495659/16563998/7b6c9986-4205-11e6-8171-c387814b0067.png)

josedlpozo committed:
add back arrow in toolbar

raulh82vlc commented on android/src/main/java/com/raulh82vlc/MoviesFinder/ui/activities/MovieDetailsActivity.java:
```diff
@@ -190,13 +191,20 @@ protected void setInitialTitle() {
     @Override
     protected void setToolbarInitialisation() {
-        mToolbar.setNavigationIcon(R.drawable.ic_back);
+        //mToolbar.setNavigationIcon(R.drawable.ic_back);
```
every code not used and which is not a descriptive comment must be removed, consequently the resource `R.drawable.ic_back` should be removed as well, as now it is never used on behalf of the default icon from the support library

raulh82vlc commented on android/src/main/java/com/raulh82vlc/MoviesFinder/ui/activities/MovieDetailsActivity.java:
```diff
@@ -190,13 +191,20 @@ protected void setInitialTitle() {
 
-        if (actionBar != null) {
-            actionBar.setDisplayHomeAsUpEnabled(true);
+        if (getSupportActionBar() != null) {
```
you are calling to `getSupportActionBar()` until 3 times, which is a bad thing in terms of performance, I highly recommend you to maintain the previous implementation of the _line 195_ and using `actionBar` instead of this getter

raulh82vlc commented on android/src/main/java/com/raulh82vlc/MoviesFinder/ui/activities/MovieDetailsActivity.java:
```diff
@@ -190,13 +191,20 @@ protected void setInitialTitle() {
+            getSupportActionBar().setHomeButtonEnabled(true);
         }
+
+        mToolbar.setNavigationOnClickListener(new View.OnClickListener() {
```
This `OnClickListener` is unnecessary here, since the back functionality is enabled by default at the toolbar when you set those previous flags to `true`, you can get rid of this bit of code.

raulh82vlc commented on android/src/main/java/com/raulh82vlc/MoviesFinder/ui/activities/MovieDetailsActivity.java:
```diff
@@ -147,6 +147,7 @@ protected void onCreate(Bundle savedInstanceState) {
             setMaterialEffect();
             getMoreDetails();
         }
+        setToolbarInitialisation();
```
thanks for this! I totally missed it

raulh82vlc commented:
thanks a lot for this PR! looks really good, as soon as you apply those changes, this would be merged

josedlpozo committed:
remove back icon

josedlpozo committed:
refactor action bar and remove onclicklistener

josedlpozo commented:
I have added some commits. I remove back icon and refactor action bar code.
I added `OnClickListener` because in some phones back arrow didn't work without it, but in most it works well.

Tell me if you think that something is missing, if not, you can close and merge this PR.

Thanks for the review!

raulh82vlc commented:
thanks @josedlpozo !
